docs/DeveloperGuide.md

```diff
@@ -6,7 +6,152 @@
+- `protected Object load() throws Exception`
+  - This method returns a simple `Object` read from the specified file. The class makes no assumption as to the
+  type of object being read. As such, further processing must be done, which will be touched on below.
+- `protected void save(Object o) throws Exception`
+  - This method saves the provided object into the specified file. It is capable of saving **any** type of Java object,
+  on the condition that it **must be serializable**. For saving complex data structures such as `ArrayLists`, it is generally
+  a good idea to also make sure that any contained objects are also serializable.
+
+Because the `load()` and `save(Object o)` methods are simple, they are generally not useful without further processing.
+As such, the `Manager` class has two other abstract methods, which extended managers must implement:
+- `protected abstract void loadData() throws Exception`
+  - Extended managers must override this method to deserialize any read objects properly. This is to ensure that
+  corrupted data is found at this phase of the programme to prevent any future errors. A good example is listed below.
+  Note that it makes use of the provided `load()` function to first read the simple `Object`, then does further
+  processing on it:
+  ```java
+    @Override
+    protected void loadData() throws Exception {
+        this.dishes = new ArrayList<>();
+        ArrayList<?> list = (ArrayList<?>) this.load();
+        for (Object object : list) {
+            this.addDish((Dish) object);
+        }
+    }
+  ```
+- `public abstract void saveData() throws Exception`
+  - This is a convenience method for doing any pre- / post-processing on any data you may want to save. Of course, you should
+  also call the `save(Object o)` method finally.
+
+Note that loading and saving is only done once. Loading is done once upon programme start-up, and saving is done
+when the user exits the menu gracefully.
+
+Here is the typical design of a manager:
+
+![](diagrams/class/manager.png)
+
+
+### Menu (dishes) Management
+
+The management of menu is handled by `DishManager`, which implements following methods:
+
+- `void printDishes()` -- print all dishes and their menu index in current menu.
+- `void addDish(Dish dish)` -- add a new dish.
+- `void deleteDish(int menuIndex)` -- delete a dish using its menu index.
+- `void setPrice(int menuIndex, double newPrice)` -- change the price of a dish using its menu index.
+- `void setName(int menuIndex, String newName)` -- change the name of a dish using its menu index.
+
+These functions are "delegated" to `ArrayList<Dish>`, which is associated with `DishManager`.
+
+For example, `printDishes()`, iterates through `dishes` and calls `get(int)` method (`setPrice(int, double)`, `setName(int, String)` also call `get(int)`):
+
+![](diagrams/sqeuence/DishManager_printDishes.png)
+
+`deleteDish(int)` calls `remove(int)`:
+
+![](diagrams/sqeuence/DishManager_deleteDish.png)
```

Comment:
Is the error supposed to be a dotted or solid line back to the users? 
![image](https://user-images.githubusercontent.com/65325165/160959658-ed9f94be-47f2-43d4-81b9-2dd01a2b04c6.png)